server/src/main/java/org/elasticsearch/index/IndexNotFoundException.java

```diff
@@ -37,7 +40,7 @@ public IndexNotFoundException(String index) {
     }
 
     public IndexNotFoundException(String index, Throwable cause) {
-        super("no such index", cause);
+        super(makeMessage(index), cause);
```

Comment:
I think it would be ok to duplicate the message like this:
```suggestion
        super("no such index ["+ index + "]", cause);
```

Reply:
we also have a convention to embed things like index names in `[ ]`